lib/Target/OpenFhePke/OpenFhePkeEmitter.cpp

```diff
@@ -277,16 +368,66 @@ LogicalResult OpenFhePkeEmitter::printOperation(func::ReturnOp op) {
   return success();
 }
 
+LogicalResult OpenFhePkeEmitter::printOperation(affine::AffineForOp op) {
+  if (!op.getConstantLowerBound() || !op.getConstantUpperBound()) {
+    return failure();
+  }
+
+  // Clone the loop's results to use as initial iter args. We must use non-const
+  // types so that the loop body can modify them.
+  for (auto i = 0; i < op.getNumRegionIterArgs(); ++i) {
+    if (failed(emitTypedAssignPrefix(op.getResults()[i], op.getLoc(), false))) {
+      return failure();
+    }
+    os << variableNames->getNameForValue(op.getOperands()[i]) << "->Clone();\n";
+    // Map the region's iter args to the result names.
+    variableNames->mapValueNameToValue(op.getRegionIterArgs()[i],
+                                       op.getResults()[i]);
+    mutableValues.insert(op.getRegionIterArgs()[i]);
+    // Map the yielded value names to the result names.
+    variableNames->mapValueNameToValue(op.getYieldedValues()[i],
+                                       op.getResults()[i]);
+    mutableValues.insert(op.getYieldedValues()[i]);
+  }
+
+  os << llvm::formatv("for (auto {0} = {1}; {0} < {2}; ++{0}) {{\n",
+                      variableNames->getNameForValue(op.getInductionVar()),
+                      op.getConstantLowerBound(), op.getConstantUpperBound());
+  os.indent();
+  for (Operation &op : *op.getBody()) {
+    if (failed(translate(op))) {
+      return op.emitOpError() << "Failed to translate for loop block";
+    }
+  }
+
+  os.unindent();
+  os << "}\n";
+  return success();
+}
+
+LogicalResult OpenFhePkeEmitter::printOperation(affine::AffineYieldOp op) {
+  // Assume all yielded loop values have already been assigned.
+  return success();
+}
+
 void OpenFhePkeEmitter::emitAutoAssignPrefix(Value result) {
-  // Use const auto& because most OpenFHE API methods would perform a copy
-  // if using a plain `auto`.
-  os << "const auto& " << variableNames->getNameForValue(result) << " = ";
+  // If the result values are iter args of a region, then avoid using a auto
+  // assign prefix.
+  if (!mutableValues.contains(result)) {
+    os << "const auto& ";
+  }
+  //  Use const auto& because most OpenFHE API methods would
+  // perform a copy if using a plain `auto`.
```

Comment:
```suggestion
  // If the result values are iter args of a region, then avoid using a auto
  // assign prefix.
  if (!mutableValues.contains(result)) {
    // Use const auto& because most OpenFHE API methods would
    // perform a copy if using a plain `auto`.
    os << "const auto& ";
  }

```